rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java

```diff
@@ -67,14 +74,44 @@ public class GitProvenance implements Marker {
     @Incubating(since = "8.9.0")
     List<Committer> committers;
 
+    @Incubating(since = "8.33.0")
+    transient Lazy<CloneUrl> cloneUrl;
+
+    public GitProvenance(UUID id,
+                         @Nullable String origin,
+                         @Nullable String branch,
+                         @Nullable String change,
+                         @Nullable AutoCRLF autocrlf,
+                         @Nullable EOL eol,
+                         @Nullable List<Committer> committers) {
+        this.id = id;
+        this.origin = origin;
+        this.branch = branch;
+        this.change = change;
+        this.autocrlf = autocrlf;
+        this.eol = eol;
+        this.committers = committers;
+        this.cloneUrl = new Lazy<>(() -> parseCloneUrl(origin));
+    }
+
+    public @Nullable CloneUrl getCloneUrl() {
+        return cloneUrl.get();
+    }
+
     /**
      * Extract the organization name, including sub-organizations for git hosting services which support such a concept,
      * from the origin URL. Needs to be supplied with the
      *
      * @param baseUrl the portion of the URL which precedes the organization
      * @return the portion of the git origin URL which corresponds to the organization the git repository is organized under
+     * @deprecated use {@link #getCloneUrl().getOrganization()} instead }
      */
+    @Deprecated
```

Comment:
I'm not sure it's necessary to deprecate since it is in widespread use. Fine to have a convenience method for organization on `CloneUrl`

Reply:
actually I think we should point back to the "old" method that does not take an argument, and deprecate this one which (again) takes the origin (with scheme).